### DISCUSSION TITLE
Extract links

### DISCUSSION THREAD

**Discussion by mgohin** [py-pdf/pypdf#2653] [Q&A]
## Explanation

We need to extract all links in our pdf documents.

To be more precise on our usage : people generate PDF with images (architects Moodboards) and on each image they add a link to the source image, so readers can have more information about the product on the image.

## Code Example

How would your feature be used? (Remove this if it is not applicable.)

```python
from pypdf import PdfReader, PdfWriter

# iterate pages
uris = page.extract_uris()
```

**Comment by stefan6419846** [comment]
Could you please provide an example file which demonstrates your current situation?

**Comment by mgohin** [comment]
Of course.
[example.pdf](https://github.com/py-pdf/pypdf/files/15337488/example.pdf)

I did manage to get links, but I'm very new to PDF syntax so I don't know if I'm on the right path

```python
reader = PdfReader("../example.pdf")
for page in reader.pages:
    for annotation in page.annotations:
        obj = annotation.get_object()
        if obj.get('/Subtype') == '/Link':
            print('Link : ', obj.get('/A').get('/URI'))
```

**Comment by pubpub-zz** [comment]
your approach is correct. Just be carefull that some times destinations are stored in named destination list.

**Comment by mgohin** [comment]
Thanks for your answers/help.

Do you have an example please ?

**Comment by pubpub-zz** [comment]
I transfer this thread to discussions
